neo4j/internal/bolt/outgoing.go

```diff
@@ -490,6 +490,27 @@ func (o *outgoing) packV(v reflect.Value) {
 		}
 	case reflect.Struct:
 		o.packStruct(v.Interface())
+	case reflect.Slice:
+		switch v.Type().Elem().Kind() {
+		case reflect.Uint8:
+			o.packer.Bytes(v.Bytes())
+			return
+		case reflect.Int, reflect.Int64, reflect.String, reflect.Float64, reflect.Interface:
```

Comment:
I think the way you're checking this case is wrong - you want to check if the type is _exactly_ `[]int` or `[]int64` or ..., not just if it's a slice with the right kind. If, say, someone declares `type ID int` and passes in a `[]ID` then it will trigger this case, but it won't get to go through the fast-path in packX. You can check type equality by pre-computing the types you care about. I think `reflect.TypeOf((*[]int)(nil)).Elem()` is the best way, until you can use `reflect.TypeFor` in `1.22`.